helm/hpcc/templates/_warnings.tpl

```diff
@@ -52,4 +52,78 @@ Pass in dict with root and warnings
   {{- $_ := set $warning "msg" (printf "The configuration contains ephemeral planes: %s" (print (uniq $match.ephemeral)) )  -}}
   {{- $_ := set $ctx "warnings" (append $ctx.warnings $warning) -}}
  {{- end -}}
+ {{- /* Warn if resources not defined for any cost component or default cpuRate being used */ -}}
+ {{- if eq .root.Values.global.cost.perCpu 0.0565000000001 -}}
+  {{- $_ := set $ctx "usingDefaultCpuCost" "true" -}}
+ {{- end -}}
+ {{- $_ := set $ctx "missingResources" list -}}
+ {{- $_ := set $ctx "defaultCpuRateComponents" list -}}
+ {{- range $component := .root.Values.eclagent -}}
+  {{- if not $component.disabled -}}
+   {{- $hasResources := include "hpcc.hasResources" (dict "resources" $component.resources) -}}
```

Comment:
question: does $component.resources cause a template problem, if the "resources" key is missing? i.e. does it need to be  something like:
```
{{- $hasResources := include "hpcc.hasResources" (dict "resources" ($component.resources | default dict)) -}}
```

(and same for $component.resource references below)

Reply:
It doesn't cause problems if the resources are missing. I tested this and it works correctly, if resources are missing.

Reply:
ok, when missing (as per our default values.yaml), I wonder what $component.resources resolves to.